pkg/plugin/sdk/deployment.go

```diff
@@ -585,6 +589,36 @@ func newDeploymentSource(source *common.DeploymentSource) DeploymentSource {
 	}
 }
 
+// Deployment represents the deployment that the stage is running. This is read-only.
+type Deployment struct {
+	// ID is the unique identifier of the deployment.
+	ID string
+	// ApplicationID is the unique identifier of the application.
+	ApplicationID string
+	// ApplicationName is the name of the application.
+	ApplicationName string
+	// PipedID is the unique identifier of the piped that is running the deployment.
+	PipedID string
+	// ProjectID is the unique identifier of the project that the application belongs to.
+	ProjectID string
+	// TriggeredBy is the name of the entity that triggered the deployment.
+	TriggeredBy string
+	// CreatedAt is the time when the deployment was created.
+	CreatedAt int64
+}
+
```

Comment:
should add comment to describe what this function does/use for (like we did in L582). The name `newXXX` is too confusing.

Reply:
@khanhtc1202 
Thanks, added comment
[9661323](https://github.com/pipe-cd/pipecd/pull/5611/commits/96613239b4d169dcd9e0c737e695990ade9c1c25)